sbol_utilities/component.py

```diff
@@ -519,3 +523,64 @@ def ed_protein(definition: str, **kwargs) -> sbol3.ExternallyDefined:
     :return: A Component object.
     """
     return sbol3.ExternallyDefined([sbol3.SBO_PROTEIN], definition, **kwargs)
+
+def ed_restriction_enzyme(name:str, **kwargs) -> sbol3.ExternallyDefined:
+    """Creates an ExternallyDefined Restriction Enzyme Component from rebase.
+
+    :param name: Name of the SBOL ExternallyDefined, used by PyDNA. Case sensitive, follow standard restriction enzyme nomenclature, i.e. 'BsaI'
+    :param kwargs: Keyword arguments of any other ExternallyDefined attribute.
+    :return: An ExternallyDefined object.
+    """
+    exec(f'from Bio.Restriction import {name}')
+    definition=f'http://rebase.neb.com/rebase/enz/{name}.html'
+    return sbol3.ExternallyDefined([sbol3.SBO_PROTEIN], definition=definition, name=name, **kwargs)
+
+def backbone(identity: str, sequence: str, dropout_location: List[int], fusion_site_length:int, linear:bool, **kwargs) -> Tuple[sbol3.Component, sbol3.Sequence]:
+    """Creates a Backbone Component and its Sequence.
+
+    :param identity: The identity of the Component. The identity of Sequence is also identity with the suffix '_seq'.
+    :param sequence: The DNA sequence of the Component encoded in IUPAC.
+    :param dropout_location: List of 2 integers that indicates the start and the end of the dropout sequence including overhangs. Note that the index of the first location is 1, as is typical practice in biology, rather than 0, as is typical practice in computer science.
+    :param fusion_site_length: Integer of the lenght of the fusion sites (eg. BsaI fusion site lenght is 4, SapI fusion site lenght is 3)
+    :param linear: Boolean than indicates if the backbone is linear, by default it is seted to Flase which means that it has a circular topology.
+    :param kwargs: Keyword arguments of any other Component attribute.
+    :return: A tuple of Component and Sequence.
+    """
+    if len(dropout_location) != 2:
+        raise ValueError('The dropout_location only accepts 2 int values in a list.')
+    backbone_component, backbone_seq = dna_component_with_sequence(identity, sequence, **kwargs)
+    backbone_component.roles.append(sbol3.SO_DOUBLE_STRANDED)  
+    dropout_location_comp = sbol3.Range(sequence=backbone_seq, start=dropout_location[0], end=dropout_location[1])
+    insertion_site_location1 = sbol3.Range(sequence=backbone_seq, start=dropout_location[0], end=dropout_location[0]+fusion_site_length)
+    insertion_site_location2 = sbol3.Range(sequence=backbone_seq, start=dropout_location[1]-fusion_site_length, end=dropout_location[1])
+      
+    if linear:
+        backbone_component.types.append(sbol3.SO_LINEAR)
+        backbone_component.roles.append(sbol3.SO_ENGINEERED_REGION)
+        open_backbone_location1 = sbol3.Range(sequence=backbone_seq, start=1, end=dropout_location[0]+fusion_site_length)
```

Comment:
You need to set the `order` properties here as well, so that open_backbone_feature knows that open_backbone_location1 comes before open_backbone_location2